File: medium/75. Sort Colors/75. Sort Colors.go
package main

import (
	"fmt"
	"slices"
)

func sortColors(nums []int) {
	count := make([]int, 0, slices.Max(nums)+1)
	fmt.Println(count)
	for _, num := range nums {
		count[num] += 1
	}

	i := 0
	for num := 0; num < len(count); num++ {
		for j := 0; j < count[num]; j++ {
			nums[i] = num
			i += 1
		}
	}
}

File: medium/75. Sort Colors/75. Sort Colors.py
from typing import List


class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # nums.sort()

        count = [0] * (max(nums) + 1)

        for num in nums:
            count[num] += 1

        i = 0
        for num, freq in enumerate(count):
            for _ in range(freq):
                nums[i] = num
                i += 1
